### DISCUSSION TITLE
Feature request: geomap action on click marker

### DISCUSSION THREAD

**Discussion by wz2b** [grafana/grafana#37757] [Dashboards, Panels & Visualizations]
I would love to have the ability to have it so that if you click on a marker on a geomap it takes some action.  One idea would be to have it change a panel variable, so the rest of the panel updates itself based on whatever you clicked.  A second idea would just be to have the query return a url field with each row, then we can put in a URL (preferably relative) that includes the new variable as a query parameter (that would be clunkier but still would get the job done).

Is this a feature anybody else is interested in?

**Comment by szerwi** [comment]
I'd also love the possibility to change the dashboard's variables using the geomap.
I plan to have multiple devices in different locations. It would be much easier to select specific device using the map rather than choosing it from the list.

**Comment by pkolyvas** [comment]
Super cool idea. We'd love to hear about more possible use cases (IE keep that feedback coming!)

**Comment by ryantxu** [comment]
Yes -- markers should support the "DataLinks" property :+1:

**Comment by niclash** [comment]
I have strong need for "Navigate to a Dashboard on click of marker". Dashboard selection would be done from a field in the query. 

Sub-feature 1; instead of 404 upon a non-existent Dashboard of that name, create the Dashboard making it easier to get it all working.
Sub-feature 2; passing any number of other fields in as Dashboard variables.
Sub-feature 3; if sub-feature 2 is implemented, then a static Dashboard should also be supported.

Use-case; We have customer with many sites (5 at the moment, hopefully 80 by end of next year) that are monitored with Grafana. GeoMap and Markers help getting good overview to people, but it would be so much better to be able to "drill down" when the overview shows an issue. At the moment, there is a Dashboard List next to the GeoMap panel, which "works" but the obvious and repeated question is "why can't I just click in the map?".

**Comment by nmarrs** [comment]
Hi @wz2b, thanks for the suggestion! It appears that we added data link support in tooltips in [this PR](https://github.com/grafana/grafana/pull/44740)

When you add a datalink it will show up in the tooltip. This should address @niclash use case of linking to other grafana dashboards via field in query.
<img width="862" alt="Screen Shot 2022-10-19 at 2 22 27 PM" src="https://user-images.githubusercontent.com/22381771/196807082-d152ff21-4224-4daf-aab2-d060a84a5f2a.png">


See [data link docs ](https://grafana.com/docs/grafana/v9.0/panels/configure-data-links/) for more info on how to use data links.

@wz2b do data links in markers address your desired use case(s)?
